quarkchain/evm/specials.py

```diff
@@ -237,26 +237,46 @@ def proc_current_mnt_id(ext, msg):
 def proc_transfer_mnt(ext, msg):
     from quarkchain.evm.messages import apply_msg
 
-    # Data must be >= 96 bytes
-    if msg.data.size < 96:
-        return 0, 0, []
-    gascost = 3
-    if msg.gas < gascost:
+    # Data must be >= 96 bytes and static call not allowed
+    if msg.data.size < 96 or msg.static:
         return 0, 0, []
     to = utils.int_to_addr(msg.data.extract32(0))
-    mnt = msg.data.extract32(32)
+    token_id = msg.data.extract32(32)
     value = msg.data.extract32(64)
     data = msg.data.extract_all(96)
+
+    # Token ID should be within range
+    if token_id > TOKEN_ID_MAX:
+        return 0, 0, []
+
+    # Doesn't allow target address to be this precompiled contract itself
+    if to == decode_hex(b"000000000000000000000000000000514b430002"):
+        return 0, 0, []
+
+    gascost = 700  # gascost of CALL
```

Comment:
I think the 700 gascost is already paid by the caller?  The essential goal is that if we implement a CALLX with the same functionality as precompiled contracts, and the cost should be close.

Reply:
i see

Reply:
done